src/client/clientmap.cpp

```diff
@@ -610,3 +659,243 @@ void ClientMap::PrintInfo(std::ostream &out)
 {
 	out<<"ClientMap: ";
 }
+
+void ClientMap::clean_mblock_shadows()
+{
+	for (auto &i : m_drawlist_shadow) {
+		MapBlock *block = i.second;
+		block->refDrop();
+	}
+	m_drawlist_shadow.clear();
+}
+
+void ClientMap::renderMapShadows(video::IVideoDriver *driver,
+		irr::video::SMaterial &material, s32 pass, irr::core::vector3df position,
+		irr::core::vector3df direction, float max_distance, bool replace_material)
+{
+	bool is_transparent_pass = pass != scene::ESNRP_SOLID;
+	std::string prefix;
+	if (is_transparent_pass)
+		prefix = "renderMap(SHADOW TRANS): ";
+	else 
+		prefix = "renderMap(SHADOW SOLID): ";
+	
+	// ultraugly hack
+	//const f32 camera_fov = m_camera_fov * 1.5f;
+
+	u32 drawcall_count = 0;
+
+	/*
+		Get all blocks and draw all visible ones
+	*/
+
+	u32 vertex_count = 0;
+
+	/*
+		Draw the selected MapBlocks
+	*/
+
+	MeshBufListList drawbufs;
+
+	for (auto &i : m_drawlist_shadow) {
+		v3s16 block_pos = i.first;
+		MapBlock *block = i.second;
+
+		// If the mesh of the block happened to get deleted, ignore it
+		if (!block->mesh)
+			continue;
+
+		/*float d = 0.0;
+		if (!isBlockInSight(block->getPos(), position - direction *4.0f,
+				    direction,
+				    camera_fov, max_distance, &d))
+			continue;*/
+		
+		/*
+			Get the meshbuffers of the block
+		*/
+		{
+			// MutexAutoLock lock(block->mesh_mutex);
```

Comment:
why is this commented out? this looks thread-unsafe

Reply:
It's a C&P from the renderMap method. It can be removed without problem.